torchaudio/datasets/commonvoice.py

```diff
@@ -0,0 +1,87 @@
+import os
+
+import torchaudio
+from torchaudio.dataset.utils import download, extract, filtering, shuffle, walk
+
+
+def load_commonvoice(fileids, tsv):
+    """
+    Load data corresponding to each COMMONVOICE fileids.
+
+    Input: path, file name identifying a row of data
+    Output: dictionary with keys from tsv along with, waveform, sample_rate
+    """
+
+    for path, fileid in fileids:
+        filename = os.path.join(path, "clips", fileid)
+        tsv = os.path.join(path, tsv)
+
+        found = False
+        with open(tsv) as tsv:
```

Comment:
You can use csvreader for this or even better write a unicode_csv_reader akin to https://github.com/pytorch/text/blob/fd31bf3722e17dfd5998b8c4f32c0431f3016d59/torchtext/utils.py#L108 (which we still need because of Python2). 

Only difference is that the delimiter is a tab instead of a comma.

Reply:
It also has support for headers and such.

Reply:
https://docs.python.org/3/library/csv.html

Reply:
thanks for pointing this out!